cardano-api/src/Cardano/Api/Governance/Poll.hs

```diff
@@ -213,3 +346,65 @@ verifyPollAnswer poll answer witness = do
         VRF.verifyVRF () vk answer (undefined, proof)
       GovernancePollWitnessColdKey vk sig ->
         verifySignedDSIGN vk answer sig
+
+
+-- ----------------------------------------------------------------------------
+-- Decoder Helpers
+--
+
+withNestedMap
+  :: Text
+  -> Word64
+  -> TxMetadata
+  -> ([(TxMetadataValue, TxMetadataValue)] -> Either DecoderError a)
+  -> Either DecoderError a
+withNestedMap lbl topLevelLabel (TxMetadata m) continueWith =
+  case Map.lookup topLevelLabel m of
+    Just (TxMetaMap values) ->
+      continueWith values
+    Nothing ->
+      Left $ DecoderErrorCustom lbl
+        ("missing expected label: " <> textShow topLevelLabel)
+    Just _ ->
+      Left $ DecoderErrorCustom lbl
+        ("malformed data; expected a key:value map")
+
+expectJust :: Text -> Maybe a -> Either DecoderError a
+expectJust lbl =
+  maybe
+    (Left (DecoderErrorCustom lbl "malformed field(s)"))
+    Right
+
+expectTextChunks :: Text -> TxMetadataValue -> Either DecoderError Text
+expectTextChunks lbl value =
+  case value of
+    TxMetaList xs ->
+      foldM expectText mempty xs
+        & maybe
+            (Left (malformedField (lbl <> "[i]") "Text"))
+            (Right . Text.Lazy.toStrict . Text.Builder.toLazyText)
+    _ ->
+      Left (malformedField lbl "List<Text>")
+ where
+  expectText acc x =
+    case x of
+      TxMetaText txt -> Just (acc <> Text.Builder.fromText txt)
+      _ -> Nothing
+
+expectWord :: Text -> Integer -> Either DecoderError Word
+expectWord lbl n
+  | n >= 0 && n < toInteger (maxBound :: Word) =
+      pure (fromInteger n)
+  | otherwise =
+      Left $ DecoderErrorCustom lbl
+        ("invalid number; must be non-negative word")
```

Comment:
There is a `hlint` warning here.  Please check for `hlint` warnings.

Reply:
Done.